haystack/nodes/prompt/invocation_layer/amazon_bedrock.py

```diff
@@ -0,0 +1,134 @@
+            "ai21.j2-ultra-v1",
+            "ai21.j2-mid-v1",
+            "cohere.command-text-v14",
+        ]:
+            return model_name_or_path
+
+    @classmethod
+    def create_session(
+        self,
+        aws_access_key_id: Optional[str] = None,
+        aws_secret_access_key: Optional[str] = None,
+        aws_session_token: Optional[str] = None,
+        aws_region_name: Optional[str] = None,
+        aws_profile_name: Optional[str] = None,
+        **kwargs,
+    ):
+        """
+        Creates an AWS Session with the given parameters.
+
+        :param aws_access_key_id: AWS access key ID.
+        :param aws_secret_access_key: AWS secret access key.
+        :param aws_session_token: AWS session token.
+        :param aws_region_name: AWS region name.
+        :param aws_profile_name: AWS profile name.
+        :raise NoCredentialsError: If the AWS credentials are not provided or invalid.
+        :return: The created AWS Session.
+        """
+        boto3_import.check()
+        return boto3.Session(
+            aws_access_key_id=aws_access_key_id,
+            aws_secret_access_key=aws_secret_access_key,
+            aws_session_token=aws_session_token,
+            region_name=aws_region_name,
+            profile_name=aws_profile_name,
+        )
+
+    def _prepare_invoke(self, prompt, **kwargs):
+        del kwargs["top_k"]
+        if self.model_name_or_path in ["amazon.titan-text-express-v1", "amazon.titan-text-lite-v1"]:
+            kwargs["stopSequences"] = kwargs["stop_words"] or []
+            kwargs["topP"] = 1 if "topP" not in kwargs else kwargs["topP"]
+            kwargs["temperature"] = 0.3 if "temperature" not in kwargs else kwargs["temperature"]
+            kwargs["maxTokenCount"] = self.max_length
+            del kwargs["stop_words"]
+            body = json.dumps({"inputText": prompt, "textGenerationConfig": {**kwargs}})
+        if self.model_name_or_path in ["ai21.j2-ultra-v1", "ai21.j2-mid-v1"]:
+            kwargs["topP"] = 1 if "topP" not in kwargs else kwargs["topP"]
+            kwargs["temperature"] = 0.3 if "temperature" not in kwargs else kwargs["temperature"]
+            kwargs["maxTokens"] = self.max_length
+            del kwargs["stop_words"]
+            body = json.dumps({"prompt": prompt, **kwargs})
+        if self.model_name_or_path in ["cohere.command-text-v14"]:
+            kwargs["temperature"] = 0.3 if "temperature" not in kwargs else kwargs["temperature"]
+            kwargs["max_tokens"] = self.max_length
+            del kwargs["stop_words"]
+            body = json.dumps({"prompt": prompt, **kwargs})
+        return body
+
+    def invoke(self, *args, **kwargs):
+        client = self.client
+        prompt = kwargs.get("prompt")
+        body = self._prepare_invoke(**kwargs)
+        r = client.invoke_model(
+            body=body, modelId=self.model_name_or_path, accept="application/json", contentType="application/json"
+        )
```

Comment:
We should catch possible errors here and raise a proper `AmazonBedrockInferenceError`. Also if it's possible to detect, let's errors for:
- model not being available 
- model access not granted